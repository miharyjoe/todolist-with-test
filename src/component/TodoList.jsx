import React from "react";

const TodoList = ({ todos, handleCheck }) => (
  <ul className="liste">
    {todos.map((todo, index) => (
      <li key={index}>
        <input
          type="checkbox"
          checked={false}
          onClick={() => handleCheck(index)}
        />
        <span>{todo}</span>
      </li>
    ))}
  </ul>
);

export default TodoList;
