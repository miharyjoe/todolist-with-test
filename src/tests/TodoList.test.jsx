import React from "react";
import { cleanup, render } from "@testing-library/react";
import TodoList from "../component/TodoList";

describe("TodoList  component", () => {
  it("snapshot: render todo list", () => {
    const todos = ["Todo 1", "Todo 2", "Todo 3"];
    const { container } = render(
      <TodoList todos={todos} handleCheck={onchange} />
    );
    expect(container).toMatchSnapshot();
  });

  it("renders a list of tasks", () => {
    const todos = ["Todo 1", "Todo 2", "Todo 3"];
    const { getByText } = render(
      <TodoList todos={todos} handleCheck={onchange} />
    );
    todos.forEach((todo) => {
      expect(getByText(todo)).toBeInTheDocument();
    });
  });
});
