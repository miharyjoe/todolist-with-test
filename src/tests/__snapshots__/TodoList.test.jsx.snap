// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList  component snapshot: render todo list 1`] = `
<div>
  <ul
    class="liste"
  >
    <li>
      <input
        type="checkbox"
      />
      <span>
        Todo 1
      </span>
    </li>
    <li>
      <input
        type="checkbox"
      />
      <span>
        Todo 2
      </span>
    </li>
    <li>
      <input
        type="checkbox"
      />
      <span>
        Todo 3
      </span>
    </li>
  </ul>
</div>
`;
